packages/block-library/src/categories/index.php

```diff
@@ -99,20 +99,26 @@ function render_block_core_categories( $attributes, $content, $block ) {
  */
 function build_dropdown_script_block_core_categories( $dropdown_id ) {
 	ob_start();
+
+	$exports = array( $dropdown_id, home_url() );
 	?>
 	<script>
-	( function() {
-		var dropdown = document.getElementById( '<?php echo esc_js( $dropdown_id ); ?>' );
-		function onCatChange() {
-			if ( dropdown.options[ dropdown.selectedIndex ].value !== -1 ) {
-				location.href = "<?php echo esc_url( home_url() ); ?>/?" + dropdown.name + '=' + dropdown.options[ dropdown.selectedIndex ].value;
+	( ( [ dropdownId, homeUrl ] ) => {
+		document.getElementById( dropdownId ).addEventListener( 'change', ( event ) => {
+			const dropdown = /** @type {HTMLSelectElement} */ ( event.target );
+			if ( dropdown.value && dropdown.value !== '-1' ) {
+				const url = new URL( homeUrl );
+				url.searchParams.set( dropdown.name, dropdown.value );
+				location.href = url.href;
 			}
-		}
-		dropdown.onchange = onCatChange;
-	})();
+		} );
+	})( <?php echo wp_json_encode( $exports, JSON_HEX_TAG | JSON_UNESCAPED_SLASHES ); ?> );
 	</script>
 	<?php
-	return wp_get_inline_script_tag( str_replace( array( '<script>', '</script>' ), '', ob_get_clean() ) );
+	return wp_get_inline_script_tag(
+		trim( str_replace( array( '<script>', '</script>' ), '', ob_get_clean() ) ) .
+		"\n//# sourceURL=" . rawurlencode( __FUNCTION__ )
+	);
```

Comment:
Note: We can't safely yet use `type=module` here because the `importmap` could be printed later at `wp_footer` in classic themes, and this would cause issues in Firefox.